ui/nuxeo-document-comments/nuxeo-document-comments-mixin.js

```diff
@@ -0,0 +1,70 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+
+const CommentsMixin = (superClass) =>
+  class extends superClass {
+    static get properties() {
+      return {
+        text: {
+          type: String,
+          value: '',
+        },
+      };
+    }
+
+    static get inputTemplate() {
+      return html`
+        <div class="input-area">
+          <paper-textarea
+            id="inputContainer"
+            placeholder="[[_computeTextLabel(level, 'writePlaceholder', null, i18n)]]"
+            value="{{text}}"
+            max-rows="[[_computeMaxRows()]]"
+            no-label-float
+            on-keydown="_checkForEnter"
+          >
+          </paper-textarea>
+          <template is="dom-if" if="[[!_isBlank(text)]]">
+            <iron-icon
+              id="submit"
+              name="submit"
+              class="main-option opaque"
+              icon="check"
+              on-tap="_submitComment"
+            ></iron-icon>
+            <nuxeo-tooltip for="submit">[[i18n('comments.submit.tooltip')]]</nuxeo-tooltip>
+            <iron-icon name="clear" class="main-option opaque" icon="clear" on-tap="_clearInput"></iron-icon>
+          </template>
+        </div>
+      `;
+    }
+
+    _clearInput() {
+      this.text = '';
+    }
+
+    _computeMaxRows() {
+      const lineHeight = parseFloat(this.getComputedStyleValue('--nuxeo-comment-line-height'));
+      const maxHeight = parseFloat(this.getComputedStyleValue('--nuxeo-comment-max-height'));
+      return Math.round((Number.isNaN(maxHeight) ? 80 : maxHeight) / (Number.isNaN(lineHeight) ? 20 : lineHeight));
+    }
+
+    _computeTextLabel(level, option, placeholder) {
+      return level === 1
+        ? this.i18n(`comments.${option}.comment`, placeholder)
+        : this.i18n(`comments.${option}.reply`, placeholder);
+    }
+
+    _isBlank(text) {
+      return !text || typeof text !== 'string' || text.trim().length === 0;
+    }
+
+    _checkForEnter(e) {
+      if (e.keyCode === 13 && e.ctrlKey && !this._isBlank(this.text)) {
+        this._submitComment();
```

Comment:
We should at least add this method to the mixin, with a comment stating that it's  an extension point and needs to be overridden by elements extending the mixin. Something like
```
// extension point: logic to submit the comment
_submitComment() {
  // we could even throw an error here
}
```